cpp/src/ray/runtime/task/native_task_submitter.cc

```diff
@@ -83,5 +86,28 @@ ObjectID NativeTaskSubmitter::SubmitActorTask(InvocationSpec &invocation,
   return Submit(invocation, task_options);
 }
 
+std::string NativeTaskSubmitter::GetFullName(bool global, const std::string &name) {
+  if (name.empty()) {
+    return "";
+  }
+  return global ? name
+                : CoreWorkerProcess::GetCoreWorker().GetCurrentJobId().Hex() + "-" + name;
+}
```

Comment:
Why write this method twice?

Reply:
GetFullName is a virtual function, one is for native mode, another is for local mode.

Reply:
Can we avoid duplicated code? Move the logic to the base class?

Reply:
Not duplicated code, the implementation are different:
native mode:
CoreWorkerProcess::GetCoreWorker().GetCurrentJobId().Hex()
local mode:
local_mode_ray_tuntime_.GetCurrentJobID().Hex()